### DISCUSSION TITLE
Assistance with case setup for Flow and Thermal analysis over a cylinder

### DISCUSSION THREAD

**Discussion by Madhav-46** [xcompact3d/Incompact3d#305] [Q&A] [answered]
Dear Incompact3D Team,

I am currently working on a project to evaluate the flow and thermal characteristics of ﬂow over a cylinder. I would greatly appreciate your guidance on setting up the case correctly. Specifically, I am looking for assistance with the following:

   1. Boundary and Initial Conditions: Suggestions on how to set the inlet, outlet, and wall boundary conditions for the thermal analysis.
   2. Energy Equation Implementation: Should I create a new source file to solve the energy equation and couple it with the Incompact3D code (e.g., Case-Cylinder-wake.f90)? If so, I would appreciate any instructions on how to achieve that.
   3. Example Cases: If available, examples or case studies related to similar simulations that I could refer.

Thank you for your assistance. I look forward to your response.

Best regards,
Madhav.

**Comment by mathrack** [comment, marked as answer]
Hello,

Have you tried to compile the code ? To run the existing input files in the [examples](https://github.com/xcompact3d/Incompact3d/tree/master/examples/Cylinder-wake) folder ? Feel free to look at the documentation. As mentioned in https://github.com/xcompact3d/Incompact3d/issues/245 the documentation does not extensively cover the total code. Still worth to check 

[https://xcompact3d.readthedocs.io/en/latest/pages/user_guide/parameters.html](https://xcompact3d.readthedocs.io/en/latest/pages/user_guide/parameters.html)
and
[https://xcompact3d.readthedocs.io/en/latest/pages/user_guide/list_files.html](https://xcompact3d.readthedocs.io/en/latest/pages/user_guide/list_files.html)

Currently there is no step-by-step guide showing the detailed modifications allowing to simulate the flow and thermal characteristics over a cylinder. Start with the current code and the existing input files for the cylinder and try to modify them. Please let us know if it goes wild and the existing documentation is not helpful.

**Comment by Madhav-46** [comment]
Hi, 

Yes, I have run the existing input files before and I also got your assistance in setting up my case-setup through my earlier query #298, I am very thankful for that. Now I want to do a thermal analysis and I need your assistance in this regard. I am looking for guidance in 

- Setting up Inlet, Outlet, Cylinder Surface Temperatures and boundary conditions
- Enabling the energy equation in Incompact3d to account for heat transfer in the simulation or modifying the source code to include thermal effects if it isn’t already included.

Best regards,
Madhav.
